tests/test_sampler.py

```diff
@@ -314,28 +313,31 @@ def test_sampling_request_callback():
 
     return_value = mock.MagicMock()
     return_value.exception = lambda *args: False
-    return_value.result = lambda *args: type('obj', (object,), {'body': 'bad_json'})()
 
-    # noinspection PyProtectedMember
-    sampler._sampling_request_callback(return_value)
-    assert error_reporter.error.call_count == 1
+    probabilistic_strategy = """
+    {
+        "strategyType":0,
+        "probabilisticSampling":
+        {
+            "samplingRate":0.002
+        }
+    }
+    """
 
-    # Strategy has changed to new probabilistic sampler
     return_value.result = lambda *args: \
-        type('obj', (object,), {'body': '{"strategyType":0,"probabilisticSampling":{"samplingRate":0.002}}'})()
+        type('obj', (object,), {'body': probabilistic_strategy})()
     # noinspection PyProtectedMember
     sampler._sampling_request_callback(return_value)
-    assert '%s' % sampler.sampler == 'ProbabilisticSampler(0.002)'
+    assert '%s' % sampler.sampler == 'ProbabilisticSampler(0.002)', 'sampler should have changed to probabilistic'
+    prev_sampler = sampler.sampler
 
-    # Strategy hasn't changed
     # noinspection PyProtectedMember
     sampler._sampling_request_callback(return_value)
-    assert '%s' % sampler.sampler == 'ProbabilisticSampler(0.002)'
+    assert prev_sampler == sampler.sampler, "strategy hasn't changed so sampler should not change"
```

Comment:
this may not be the correct comparison, doesn't Python treat `==` as `__eq__`? We need to compare references.